sdk/storage/azure-storage-blob/tests/test_blob_client.py

```diff
@@ -711,4 +711,24 @@ def test_closing_pipeline_client_simple(self, **kwargs):
                 self.account_url(storage_account_name, "blob"), credential=storage_account_key, container_name='foo', blob_name='bar')
             service.close()
 
+    @BlobPreparer()
+    def test_validate_empty_blob(self, **kwargs):
```

Comment:
Three asks for the test:
- Would you mind moving this test to `test_common_blob.py`? This file is more for testing the BlobClient rather than upload. I think you'll have to change the test a little since that file has not yet migrated to test proxy.
- Can you record the test?
- Can you add the same test to `test_common_blob_async.py` too? We usually duplicate all of tests for sync and async. Obviously, you'll have to make a couple updates to support async. Please also record this one.